File: 0122_best_time_to_buy_and_sell_stock_II/max_profit.java
class Solution {
    public int maxProfit(int[] prices) {
        // Time: O(n)
        // Space: O(1)
        int res = 0;
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] > prices[i - 1]) {
                res += prices[i] - prices[i - 1];
            }
        }
        return res;
    }
}

File: 0122_best_time_to_buy_and_sell_stock_II/max_profit.rb
# @param {Integer[]} prices
# @return {Integer}
def max_profit(prices)
    # Time: O(n)
    # Space: O(1)
    res = 0
    return res if prices.length < 2
    prices.each_with_index do |p, i|
      res += (prices[i + 1] - p) if prices[i + 1] && prices[i + 1] > p
    end  
    res
end